### PR TITLE
[Easy][c10d] Minor fixes

### PR DESCRIPTION
**Pull request by awgu**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#73318 [Easy][c10d] Minor fixes**
* #73299 [DDP][BE] Fix clang-tidy
* #73237 [DDP][BE] Remove bucket replicas

Differential Revision: [D34431929](https://our.internmc.facebook.com/intern/diff/D34431929)

### CODE REVIEW

File: torch/csrc/distributed/c10d/Utils.cpp [hunk 1]
```diff
@@ -72,15 +72,10 @@ std::vector<at::Tensor> getTensorShapes(
...
-    // Need to clone here otherwise the shapesVec.data() memory is not copied
-    // and can be released under the hood.
-    at::Tensor shapesTensor = at::from_blob(
-                                  shapesVec.data(),
-                                  {shapes_size},
-                                  at::TensorOptions().dtype(at::kLong))
-                                  .clone();
+    // Use `at::tensor()` to copy the data underlying `sizes()` since it may be
+    // released elsewhere.
+    at::Tensor shapesTensor =
```
Comment (awgu) [inline torch/csrc/distributed/c10d/Utils.cpp:77]: `at::tensor()` should make a copy: https://discuss.pytorch.org/t/can-i-initialize-tensor-from-std-vector-in-libtorch/33236/4

File: torch/csrc/distributed/c10d/ProcessGroupWrapper.cpp [hunk 2]
```diff
@@ -143,12 +143,12 @@ std::ostream& operator<<(
     std::vector<std::string> dtype_strs;
     std::vector<std::string> device_type_strs;
     for (const auto& tensor_dtype : collective_fingerprint.tensor_dtypes_) {
-      dtype_strs.push_back(
+      dtype_strs.emplace_back(
```
Comment (awgu) [inline torch/csrc/distributed/c10d/ProcessGroupWrapper.cpp:146]: clang-tidy

File: torch/csrc/distributed/c10d/Utils.cpp (hunk 1 above)
Comment (awgu) [reply torch/csrc/distributed/c10d/Utils.cpp:77]: Previously, `tensor.sizes().vec()` copies the data (underlying `tensor.sizes()`) once, and then `from_blob().clone()` copies the data a second time. This new approach _should_ save one copy.

**Comment by awgu** [comment]
@awgu has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34431929).

File: torch/csrc/distributed/c10d/Utils.cpp (hunk 1 above)
Comment (mrshenli) [reply torch/csrc/distributed/c10d/Utils.cpp:77]: good catch!

**Review by mrshenli — APPROVED** [review]

**Pull request closed without merging**